socorro/lib/transaction.py

```diff
@@ -48,9 +48,10 @@ def __call__(self, function, *args, **kwargs):
                 connection.commit()
                 return result
             except NonFatalException:
+                excinfo = sys.exc_info()
                 # This is a non-fatal exception so we don't need to reconnect
                 connection.rollback()
-                raise
+                reraise(*excinfo)
```

Comment:
If `connection.rollback()` throws an error, then `raise` does the wrong thing. So this fixes it to capture the original error and reraise that.